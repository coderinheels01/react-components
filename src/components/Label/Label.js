import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

export const LabelWrapper = styled.label`
  display: block;
`;
export const LabelRequired = styled.span`
  color: red;
`;

/** Label with required field display and  html*/
const Label = ({ label, htmlFor, required }) => (
  <LabelWrapper htmlFor={htmlFor}>
    {label}
    {required && <LabelRequired> * </LabelRequired>}
  </LabelWrapper>
);

Label.propTypes = {
  /** Label text */
  label: PropTypes.string.isRequired,

  /** HTML id for associated input*/
  htmlFor: PropTypes.string.isRequired,

  /** Display red asterisk if required is true */
  required: PropTypes.bool
};

export default Label;
