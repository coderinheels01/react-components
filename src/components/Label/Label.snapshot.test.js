import React from "react";
import renderer from "react-test-renderer";
import Label from "./Label";

describe("Label snapshot test", () => {
  let props;
  let labelTree;

  describe("when required props is false", () => {
    beforeEach(() => {
      props = {
        label: "test label ",
        htmlFor: "test-html-id",
        required: false
      };
      labelTree = renderer.create(<Label {...props} />);
    });
    it("should not create 'LabelRequired' ", () => {
      expect(labelTree).toMatchSnapshot();
    });
    afterEach(() => {
      labelTree.unmount();
    });
  });

  describe("when required props is true", () => {
    beforeEach(() => {
      props = {
        label: "test label ",
        htmlFor: "test-html-id",
        required: true
      };
      labelTree = renderer.create(<Label {...props} />);
    });
    it("should not create 'LabelRequired' ", () => {
      expect(labelTree).toMatchSnapshot();
    });
    afterEach(() => {
      labelTree.unmount();
    });
  });
});
