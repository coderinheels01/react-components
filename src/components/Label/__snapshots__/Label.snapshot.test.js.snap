// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label snapshot test when required props is false should not create 'LabelRequired'  1`] = `
<label
  className="sc-bdfBwQ gaSFkq"
  htmlFor="test-html-id"
>
  test label 
</label>
`;

exports[`Label snapshot test when required props is true should not create 'LabelRequired'  1`] = `
<label
  className="sc-bdfBwQ gaSFkq"
  htmlFor="test-html-id"
>
  test label 
  <span
    className="sc-gsTCUz wIMML"
  >
     * 
  </span>
</label>
`;
